### DISCUSSION TITLE
Efficiently Converting a jsi::Array to a C++ std::vector Without Explicit Loops

### DISCUSSION THREAD

**Discussion by shivamst11** [facebook/hermes#1478] [General]
Hi everyone,

I'm working on a project where I need to convert a jsi::Array  into a C++ std::vector. The array could potentially be quite large, so I'm concerned about the performance implications of using explicit loops for this conversion.

I'm looking for a best practices to handle this situation. Any insights or code snippets would be greatly appreciated!

Thank you!

**Comment by tmikov** [comment]
Hi! A jsi::Array can contain any kind of elements, including objects managed by the GC, so it is generally not safe to copy them into a std::vector.  It could work if the array contained **only numbers**, but that is a very specific case, so there isn't an optimized API for it.

However, if you need only numbers, I think it would be much more efficient to use an ArrayBuffer and copy from the underlying data pointer directly.

**Reply by shivamst11** [reply]
Thanks for your response.

now i am passing ArrayBuffer from js to module and using it, and it solved my problem.
